core/arch/arm/kernel/pseudo_ta.c

```diff
@@ -90,11 +91,22 @@ static TEE_Result copy_in_param(struct tee_ta_session *s __maybe_unused,
 		case TEE_PARAM_TYPE_MEMREF_INOUT:
 			if (!validate_in_param(s, param->u[n].mem.mobj))
 				return TEE_ERROR_BAD_PARAMETERS;
-
 			va = mobj_get_va(param->u[n].mem.mobj,
 					 param->u[n].mem.offs);
-			if (!va)
-				return TEE_ERROR_BAD_PARAMETERS;
+			if (!va) {
+				TEE_Result res;
+
+				res  = mobj_reg_shm_map(param->u[n].mem.mobj);
+				if (res)
+					return res;
+				did_map[n] = true;
```

Comment:
move `did_map[n] = true;` after `va` is validated below ? can get_va fail is mapping succeeded ?

Reply:
Currently I don't think it can fail. But since `mobj_reg_shm_map()` succeeded we need to call `mobj_reg_shm_unmap()` regardless if `mobj_get_va()` succeeds or not.

Reply:
ok